packages/nextjs/app/_components/CollateralGraph.tsx

```diff
@@ -46,68 +46,54 @@ const CollateralGraph = () => {
+  const sortedEvents = combinedEvents.sort((a, b) => Number((a as any).blockNumber - (b as any).blockNumber));
 
-  const ratioData = sortedEvents.reduce((acc, event, idx) => {
-    const collateralAdded = event.eventName === "CollateralAdded" ? event.args.amount : 0n;
-    const collateralWithdrawn = event.eventName === "CollateralWithdrawn" ? event.args.amount : 0n;
-    const price = event.args.price || 3333000000000000000000n;
-    const debtAdded = event.eventName === "Transfer" ? getDebtFromTransferEvent(event) : 0n;
+  const ratioData = sortedEvents.reduce<Array<{ name: number; ratio: number; collateral: bigint; debt: bigint }>>(
+    (acc, event) => {
+      try {
+        const collateralAdded = event.eventName === "CollateralAdded" ? BigInt(event.args.amount || 0) : 0n;
+        const collateralWithdrawn = event.eventName === "CollateralWithdrawn" ? BigInt(event.args.amount || 0) : 0n;
+        const price =
+          event.eventName === "CollateralAdded" || event.eventName === "CollateralWithdrawn"
+            ? BigInt(event.args.price || 0)
+            : 0n;
+        const debtAdded = event.eventName === "Transfer" ? getDebtFromTransferEvent(event) : 0n;
 
-    const prevCollateral = acc[idx - 1]?.collateral || 0n;
-    const prevDebt = acc[idx - 1]?.debt || 0n;
+        const prevCollateral = acc[acc.length - 1]?.collateral || 0n;
+        const prevDebt = acc[acc.length - 1]?.debt || 0n;
 
-    const collateralInEth = prevCollateral + collateralAdded - collateralWithdrawn;
-    const ethPriceInStable = BigInt(Math.round(Number(formatEther(price))));
-    const collateralInStable = collateralInEth * ethPriceInStable;
-    const debt = prevDebt + debtAdded;
-    const ratio = Number(collateralInStable || 1) / Number(debt || collateralInStable || 1);
+        const collateral = prevCollateral + (collateralAdded - collateralWithdrawn) * (price ? price : 0n);
+        const debt = prevDebt + debtAdded;
 
-    return [
-      ...acc,
-      {
-        name: event.blockData?.number || 0,
-        ratio: Number.isFinite(ratio) ? ratio : 1,
-        collateral: collateralInEth,
-        debt: debt,
-      },
-    ];
-  }, []);
+        // Avoid division by zero and ensure proper number conversion
+        const ratio = debt === 0n ? (collateral === 0n ? 1 : Number(collateral)) : Number(collateral) / Number(debt);
+
+        return [
+          ...acc,
+          {
+            name: (event as any).blockNumber || 0,
+            ratio: Number.isFinite(ratio) ? ratio : 1,
+            collateral: collateral,
+            debt: debt,
+          },
+        ];
+      } catch (error) {
+        console.error("Error processing event:", error);
+        return acc;
+      }
+    },
+    [],
+  );
 
   return (
     <div className="card bg-base-100 w-96 shadow-xl">
       <div className="card-body h-96 w-96">
-        <h2 className="card-title">Total Collateral/Debt Ratio</h2>
```

Comment:
I ended up changing this title to be `Total Collateral/Debt Ratio` because I thought it would be a bit easier to understand. What do you think @Astronaut828 ?